command/server.go

```diff
@@ -986,8 +986,8 @@ CLUSTER_SYNTHESIS_COMPLETE:
 			for _, name := range list {
 				path := filepath.Join(f.Name(), name)
 				if err := c.addPlugin(path, init.RootToken, core); err != nil {
-					c.UI.Error(fmt.Sprintf("Error enabling plugin %s: %s", name, err))
-					return 1
+					pluginsNotLoaded = append(pluginsNotLoaded, name)
```

Comment:
Maybe this should be limited to just those failing due to "unable to determine plugin type", and for other errors, emit the error and quit as normal. Otherwise if you have a faulty plugin you'll not see the error info.

Reply:
What a bout simply storing and printing the errors, and continuing for all plugin registration error cases? If we want to repro the error case we'd just run the registration manually.

Reply:
That sounds OK. Showing those errors somehow was my main concern.

Reply:
This might not be as straight forward since the error value is multi-line and quite long so having it next to the plugin name can be messy.

Reply:
The error coming back looks something like this which can be hard to log if we have (multiple errors) x (multiple failing plugins):
```
* 1 error occurred:
        * unsupported operation
```

I kept it simple and resorted to checking for a specific error, continue on that, and failing for the rest as you initially suggested.